### ISSUE TITLE
tests fail on 32bit (i686 and armv7hl)

### ISSUE THREAD

**Issue by carlwgeorge** [transip/gotransip#11]
## (Problem) Description

I'm packaging this library for Fedora, and I discovered that the tests fail on 32bit architectures.  Specifically they failed on [i686](https://kojipkgs.fedoraproject.org//work/tasks/9971/32889971/build.log) and [armv7hl](https://kojipkgs.fedoraproject.org//work/tasks/9973/32889973/build.log).

## Steps to reproduce

1. use a 32bit architecture
2. `cd vps`
3. `go test` (this may or may not need Fedora's build flags such as `-buildmode pie` to reproduce, I can't confirm as I don't have this hardware and can only trigger the error in Fedora build)

## Expected behaviour

all tests pass

## Actual behaviour

```
Testing: "/builddir/build/BUILD/gotransip-5.8.1/_build/src/github.com/transip/gotransip/vps"
+ GOPATH=/builddir/build/BUILD/gotransip-5.8.1/_build:/usr/share/gocode
+ go test -buildmode pie -compiler gc -ldflags '-extldflags '\''-Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '\'''
--- FAIL: TestGetTrafficInformationForVps (0.00s)
    api_test.go:213: 
        	Error Trace:	api_test.go:213
        	Error:      	Not equal: 
        	            	expected: 2483004356
        	            	actual  : 2147483647
        	Test:       	TestGetTrafficInformationForVps
    api_test.go:214: 
        	Error Trace:	api_test.go:214
        	Error:      	Not equal: 
        	            	expected: 4380854877
        	            	actual  : 2147483647
        	Test:       	TestGetTrafficInformationForVps
    api_test.go:215: 
        	Error Trace:	api_test.go:215
        	Error:      	Not equal: 
        	            	expected: 1073741824000
        	            	actual  : 2147483647
        	Test:       	TestGetTrafficInformationForVps
--- FAIL: TestGetPooledTrafficInformation (0.00s)
    api_test.go:231: 
        	Error Trace:	api_test.go:231
        	Error:      	Not equal: 
        	            	expected: 128356924407
        	            	actual  : 2147483647
        	Test:       	TestGetPooledTrafficInformation
    api_test.go:232: 
        	Error Trace:	api_test.go:232
        	Error:      	Not equal: 
        	            	expected: 200639805776
        	            	actual  : 2147483647
        	Test:       	TestGetPooledTrafficInformation
    api_test.go:233: 
        	Error Trace:	api_test.go:233
        	Error:      	Not equal: 
        	            	expected: 16106127360000
        	            	actual  : 2147483647
        	Test:       	TestGetPooledTrafficInformation
FAIL
```

**Comment by skoef** [comment]
I get why this is failing, but I'm not sure how to fix it. Since these are integers higher than maxint32, we could either skip the test on 32bit platforms, on 32bit test against maxint32 instead of the actual value or just lower the counters in the shipped XML output for `TestGetPooledTrafficInformation`. Each solution will make tests pass on a 32bit platform, but will probably cause weird behaviour when you would ever use the `GetPooledTrafficInformation` function.

What would you like to see happening?

**Comment by carlwgeorge** [comment]
Oh yeah, I should have recognized 2,147,483,647.  I've seen this problem before with other software.  Skipping the test on 32bit would be fine by me.

**Comment by skoef** [comment]
@carlwgeorge it turned out the integers weren't properly parsed, could you test this branch and see if it solves your problem?

**Comment by carlwgeorge** [comment]
I applied #14 as a patch in the package, rebuilt locally, and the tests now pass on 32bit platforms.  Thanks!

I already got the [package](https://src.fedoraproject.org/rpms/golang-github-transip-gotransip) added, [skipping the tests on 32bit](https://src.fedoraproject.org/rpms/golang-github-transip-gotransip/c/4d9a0e7a41c42925f9457fbecb40858093eae388).  I could either add the patch and run the tests everywhere, or just wait for the next release that includes this fix then package that version.  I'm fine either way, really just curious if a new release is planned anytime soon.
